pkg/controller/core/localqueue_controller.go

```diff
@@ -74,22 +78,26 @@ func (r *LocalQueueReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling LocalQueue")
 
-	// Shallow copy enough for now.
-	oldStatus := queueObj.Status
-
-	pending, err := r.queues.PendingWorkloads(&queueObj)
+	var cq kueue.ClusterQueue
+	queueIsInactiveMsg := "Can't submit new workloads to clusterQueue"
+	err := r.client.Get(ctx, client.ObjectKey{Name: string(queueObj.Spec.ClusterQueue)}, &cq)
 	if err != nil {
-		r.log.Error(err, "Failed to retrieve localQueue status")
-		return ctrl.Result{}, err
+		if apierrors.IsNotFound(err) {
+			err = r.UpdateStatusIfChanged(ctx, &queueObj, metav1.ConditionFalse, "ClusterQueueDoesNotExist", queueIsInactiveMsg)
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		} else {
+			r.log.Error(err, "Failed to get a clusterQueue that localQueue references",
```

Comment:
Doesn't the manager log the returned error? If so, maybe wrapping the error with `%w` could be useful.

Reply:
I removed the log since the manager log reports a similar message.